### PR TITLE
Update plugin version in the 2.x branch

### PR DESCRIPTION
**Pull request by Emily-Jiang**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,29 +46,28 @@
...
         <version.microprofile.build-tools>1.2</version.microprofile.build-tools>
+        <version.plugin.compiler>3.13.0</version.plugin.compiler>
+        <version.plugin.resources>3.3.1</version.plugin.resources>
+        <version.plugin.jar>3.4.1</version.plugin.jar>
+        <version.plugin.dependency>3.7.0</version.plugin.dependency>
+        <version.plugin.source>3.3.1</version.plugin.source>
+        <version.plugin.javadoc>3.7.0</version.plugin.javadoc>
+        <version.plugin.helper>3.6.0</version.plugin.helper>
+        <version.plugin.impsort>1.10.0</version.plugin.impsort>
+        <version.plugin.formatter>2.24.1</version.plugin.formatter>
```
Comment (Azquelt) [inline pom.xml:57]: The formatter plugin now requires Java 17. We will need to update our CI jobs to build with Java 17 (though we can still target earlier Java versions)

```
[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.24.1:format (default) on project microprofile-openapi-parent: Execution default of goal net.revelc.code.formatter:formatter-maven-plugin:2.24.1:format failed: Unable to load the mojo 'format' in the plugin 'net.revelc.code.formatter:formatter-maven-plugin:2.24.1' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: JVMCFRE199E bad major version 61.0 of class=net/revelc/code/formatter/FormatterMojo, the maximum supported major version is 55.0; offset=6
```

File: pom.xml [hunk 2]
```diff
@@ -46,29 +46,28 @@
...
-        <version.plugin.helper>3.3.0</version.plugin.helper>
-        <version.plugin.impsort>1.8.0</version.plugin.impsort>
+        <version.plugin.compiler>3.13.0</version.plugin.compiler>
+        <version.plugin.resources>3.3.1</version.plugin.resources>
+        <version.plugin.jar>3.4.1</version.plugin.jar>
+        <version.plugin.dependency>3.7.0</version.plugin.dependency>
+        <version.plugin.source>3.3.1</version.plugin.source>
+        <version.plugin.javadoc>3.7.0</version.plugin.javadoc>
+        <version.plugin.helper>3.6.0</version.plugin.helper>
+        <version.plugin.impsort>1.10.0</version.plugin.impsort>
```
Comment (Azquelt) [inline pom.xml:56]: This version of impsort also requires Java 17.

~~With this line reverted, it does build with Java 11, so the rest should be fine.~~ Spoke too soon, there are other Java 17 dependencies which don't become apparent until you build something else which has this as its parent.

File: pom.xml [hunk 3]
```diff
@@ -46,29 +46,28 @@
...
         <version.plugin.formatter>2.22.0</version.plugin.formatter>
+        <version.plugin.checkstyle>3.4.0</version.plugin.checkstyle>
+        <version.plugin.rat>0.16.1</version.plugin.rat>
+        <version.plugin.bnd>7.0.0</version.plugin.bnd>
```
Comment (Azquelt) [inline pom.xml:60]: BND 7 requires Java 17.

With impsort, formatter and bnd reverted to their old versions, I can build this project and the MP OpenAPI spec with Java 11.

Comment (Emily-Jiang) [reply pom.xml:60]: Let's update the build to use Java 17 to build as the previous version of impsort has CVEs.

**Review by Azquelt — APPROVED** [review]

**Pull request merged**